source/MaterialXCore/Types.cpp

```diff
@@ -62,6 +63,30 @@ template <> Matrix33 MatrixN<Matrix33, float, 3>::getAdjugate() const
         _arr[0][0]*_arr[1][1] - _arr[1][0]*_arr[0][1]);
 }
 
+template <> Matrix33 MatrixN<Matrix33, float, 3>::createTranslation(Matrix33::RowArray& v)
```

Comment:
Since the dimension of these input vectors is known, I would recommend that we use the more precise Vector3 type, so that no casting is required in client code.  Ideally, we should only use the lower-level MatrixN::RowArray when we have no access to the dimension of a vector at compile time (e.g. in the index operators for MatrixN).

Reply:
I did this replacement after the change which removed the vector type as input to the I can add that back into the template or move these to specializations. I think the latter is better since scale vector should be row-dimension-1. Currently there's this extra dimension which is unused.